pvlib/iotools/psm3.py

```diff
@@ -0,0 +1,133 @@
+        in decimal degrees, between -180 and 180, east is positive
+    tmy : str
+        PSM3 parameter "name", see notes below for options, default ``'tmy'``
+    interval : integer
+        interval size in minutes, can be only either 30 or 60, default 60
+
+    Returns
+    -------
+    headers : dict
+        meta data from NREL PSM-3 about the record, see notes for fields
+    data : pandas.DataFrame
+        timeseries data from NREL PSM3
+
+    Raises
+    ------
+    requests.HTTPError
+
+    Notes
+    -----
+    The ``tmy`` parameter 'names' must be a single value from the following
+    list:
+
+        ['1998', '1999', '2000', '2001', '2002', '2003', '2004', '2005',
+         '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
+         '2014', '2015', '2016', '2017', 'tmy', 'tmy-2016', 'tmy-2017',
+         'tdy-2017', 'tgy-2017']
+
+    The return is a tuple with two items. The first item is a header with meta
+    data from NREL PSM3 about the record containing the following fields:
+
+    * Source
+    * Location ID
+    * City
+    * State
+    * Country
+    * Latitude
+    * Longitude
+    * Time Zone
+    * Elevation
+    * Local Time Zone
+    * Dew Point Units
+    * DHI Units
+    * DNI Units
+    * GHI Units
+    * Temperature Units
+    * Pressure Units
+    * Wind Direction Units
+    * Wind Speed
+    * Surface Albedo Units
+    * Version
+
+    See Also
+    --------
+    pvlib.iotools.read_tmy2, pvlib.iotools.read_tmy3
+
+    References
+    ----------
+    [1] `NREL Developer Network - Physical Solar Model (PSM) v3
+        <https://developer.nrel.gov/docs/solar/nsrdb/psm3_data_download/>`_
+    [2] `NREL National Solar Radiation Database (NSRDB)
+        <https://nsrdb.nrel.gov/>`_
+    """
+    longitude = ('%9.4f' % longitude).strip()
+    latitude = ('%8.4f' % latitude).strip()
+    params = {
+        'api_key': 'DEMO_KEY',
+        'full_name': 'Sample User',
+        'email': 'sample@email.com',
+        'affiliation': 'Test Organization',
+        'reason': 'Example',
+        'mailing_list': 'true',
+        'wkt': 'POINT(%s %s)' % (longitude, latitude),
+        'names': tmy,
+        'attributes':  ','.join(ATTRIBUTES),
+        'leap_day': 'false',
+        'utc': 'false',
+        'interval': interval
+    }
+
+    s = requests.get(URL, params=params)
+    if s.ok:
+        f = io.StringIO(s.content.decode('utf-8'))
+        x = csv.reader(f, delimiter=',', lineterminator='\n')
+        y = list(x)
+        z = dict(zip(y[0], y[1]))
+        # in USA all timezones are integers
+        z['Local Time Zone'] = int(z['Local Time Zone'])
+        z['Time Zone'] = int(z['Time Zone'])
+        z['Latitude'] = float(z['Latitude'])
+        z['Longitude'] = float(z['Longitude'])
+        z['Elevation'] = int(z['Elevation'])
+        tz = pytz.timezone('Etc/GMT%+d' % -z['Time Zone'])
+        w = pd.DataFrame(y[3:], columns=y[2], dtype=float)
+        w.Year = w.Year.apply(lambda yr: int(yr))
```

Comment:
I don't understand why you specify dtype=float above and then set the columns to be integers here. Wouldn't it be better to specify the correct dtype when creating the dataframe?

If we keep this, I think `.astype(int)` is preferable to the element by element apply.